### PR TITLE
Fix platforms for FMU export

### LINKED ISSUE #8168: FMU export options "none", "dynamic", "static" are broken.
In "Tools -> Options -> FMI" one can choose the type of "none", "dynamic", "static". The problem is they all do the same, generate a ''static" FMU no matter what. 
So either the options should be taken away or they should be fixed. 

The documentation also mentions "with [dynamically|statically] linked runtime for current platform." But what does this mean when one has chosen several target platforms. Will the win64 target include a link to some OMC win library even if compiled on a linux machine? This should be clarified. 

### Version
OpenModelica 1.19.0~dev-388-gb35aa9c on Ubuntu Linux 20.04

### PR DESCRIPTION
**Pull request by adeas31**
### Related Issues

Fixes #8168, #6009

### Purpose

Improve the FMU export platforms.

### Approach

Added a flat list of platforms. If no platform is selected then the source-code FMU is generated.

### CODE REVIEW

**Review by AnHeuermann — COMMENTED** [review]
One problem with the options for Docker is that OMEdit will freeze (not responding) for several minutes when trying to download a docker image.

Any idea how to handle that?
The messages from the API are only displayed after `buildModelFMU` is finished.

File: OMEdit/OMEditLIB/Options/OptionsDialog.cpp [hunk 1]
```diff
@@ -4951,6 +4954,30 @@ FMIPage::FMIPage(OptionsDialog *pOptionsDialog)
...
+  QCheckBox *pNativePlatformCheckBox = new QCheckBox("Native");
+  pNativePlatformCheckBox->setChecked(true);
+  pNativePlatformCheckBox->setProperty(Helper::fmuPlatformNamePropertyId, "static");
+  pPlatformsLayout->addWidget(pNativePlatformCheckBox);
+  // docker platforms
+  QStringList dockerPlarforms;
+  dockerPlarforms << "x86_64-linux-gnu docker run docker.openmodelica.org/build-deps:v1.13"
+                  << "i686-linux-gnu docker run docker.openmodelica.org/build-deps:v1.13-i386"
+                  << "x86_64-w64-mingw32 docker run docker.openmodelica.org/msyscross-omsimulator:v2.0"
+                  << "i686-w64-mingw32 docker run docker.openmodelica.org/msyscross-omsimulator:v2.0";
```
Comment (sjoelund) [inline OMEdit/OMEditLIB/Options/OptionsDialog.cpp:4975]: Why is the arm platform missing? This is arm32 (our image)/arm64 (multiarch/crossbuild from docker hub):
```
arm-linux-gnueabihf docker run docker.openmodelica.org/armcross-omsimulator:v2.0
aarch64-linux-gnu docker run multiarch/crossbuild
```
In order to speed up docker responsiveness, you could add `docker run --pull=never` to not wait for 10 minutes while docker downloads the container.
I would suggest:
```c++
<< "x86_64-linux-gnu docker run --pull=never multiarch/crossbuild"
<< "i686-linux-gnu docker run --pull=never multiarch/crossbuild"
<< "x86_64-w64-mingw32 docker run --pull=never multiarch/crossbuild"
<< "i686-w64-mingw32 docker run --pull=never multiarch/crossbuild"
<< "arm-linux-gnueabihf docker run --pull=never multiarch/crossbuild"
<< "aarch64-linux-gnu docker run --pull=never multiarch/crossbuild"
```
We could have both options (that we test) and the more up to date multiarch as well

**Comment by AnHeuermann** [comment]
And because of https://github.com/OpenModelica/OpenModelica/issues/7458 the Docker options are not usable on Windows at the moment. But that's of course a different issue and I would merge this nonetheless, after the freezing issue is adressed.

**Comment by sjoelund** [comment]
> One problem with the options for Docker is that OMEdit will freeze (not responding) for several minutes when trying to download a docker image.

See my comment. You can add `--pull=never` and get an error-message instead.

**Comment by AnHeuermann** [comment]
Mhh... I don't really like that. It worked just fine, only that the user was not aware why OMEdit needed so much time.
We can of course run some docker commands before building a FMU to print a info if an image is downlaaded.

With `--pull=never` I'll get:
```bash
[8] 15:48:12 Scripting Error
Error opening file: C:/OMDev/tools/msys/tmp/OpenModelica/OMEdit/Modelica.Electrical.Analog.Examples.ChuaCircuit/Modelica_Electrical_Analog_Examples_ChuaCircuit.fmutmp/resources/"aarch64-linux-gnu.log: No such file or directory.
```

**Comment by sjoelund** [comment]
But then we need to parse that command to guess what the image name is. It would sort of make sense for OMEdit since you end up locking the GUI for 10 minutes or so? It's much better if it's handled externally.

**Review by sjoelund — APPROVED** [review]
I think this would be good to test on the different platforms later, yes. It's not the same docker images we try out and maybe multiarch/crossbuild is too recent for some users. But at least it should be easy to add custom images.

**Review by AnHeuermann — APPROVED** [review]
Should solve the confusion about static/dynamic FMUs and enables users to use the cool docker features.

**Pull request merged**